java/client/src/org/openqa/selenium/io/TemporaryFilesystem.java

```diff
@@ -145,6 +145,10 @@ boolean shouldReap() {
   }
 
   public boolean deleteBaseDir() {
-    return baseDir.delete();
+    boolean wasDeleted = baseDir.delete();
+    if (wasDeleted) {
```

Comment:
I was just trying to check on what java is doing. It appears that delete could return false if the path was already deleted. Which we would want to remove the shutdown hook at that point too.

Maybe change this line to if (!baseDir.exists())